Description: workaround tsc update
 * Fixes also #1028888
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2023-01-15

--- a/sindresorhus-is/source/index.ts
+++ b/sindresorhus-is/source/index.ts
@@ -232,6 +232,7 @@
 is.arrayBuffer = isObjectOfType<ArrayBuffer>('ArrayBuffer');
 is.sharedArrayBuffer = isObjectOfType<SharedArrayBuffer>('SharedArrayBuffer');
 is.dataView = isObjectOfType<DataView>('DataView');
+// @ts-ignore
 is.enumCase = <T = unknown>(value: unknown, targetEnum: T) => Object.values(targetEnum).includes(value as string);
 
 is.directInstanceOf = <T>(instance: unknown, class_: Class<T>): instance is T => Object.getPrototypeOf(instance) === class_.prototype;
--- a/source/as-promise/create-rejection.ts
+++ b/source/as-promise/create-rejection.ts
@@ -13,6 +13,7 @@
 					}
 				}
 			} catch (error_) {
+				// @ts-ignore
 				error = error_;
 			}
 		}
--- a/source/as-promise/index.ts
+++ b/source/as-promise/index.ts
@@ -9,6 +9,7 @@
 	HTTPError,
 	CancelError
 } from './types';
+// @ts-ignore
 import parseBody from './parse-body';
 import Request from '../core';
 import proxyEvents from '../core/utils/proxy-events';
@@ -80,6 +81,7 @@
 						response.body = rawBody.toString();
 
 						if (isResponseOk(response)) {
+							// @ts-ignore
 							request._beforeError(error);
 							return;
 						}
@@ -119,7 +121,9 @@
 							return promise;
 						});
 					}
+				// @ts-ignore
 				} catch (error) {
+					// @ts-ignore
 					request._beforeError(new RequestError(error.message, error, request));
 					return;
 				}
--- a/source/as-promise/parse-body.ts
+++ b/source/as-promise/parse-body.ts
@@ -26,6 +26,7 @@
 			name: 'Error'
 		}, response);
 	} catch (error) {
+		// @ts-ignore
 		throw new ParseError(error, response);
 	}
 };
--- a/source/core/index.ts
+++ b/source/core/index.ts
@@ -11,7 +11,6 @@
 import CacheableLookup from 'cacheable-lookup';
 import CacheableRequest = require('cacheable-request');
 import decompressResponse = require('decompress-response');
-// @ts-expect-error Missing types
 import http2wrapper = require('http2-wrapper');
 import lowercaseKeys = require('lowercase-keys');
 import ResponseLike = require('responselike');
@@ -258,6 +257,7 @@
 export type ParseJsonFunction = (text: string) => unknown;
 export type StringifyJsonFunction = (object: unknown) => string;
 
+// @ts-ignore
 interface RealRequestOptions extends https.RequestOptions {
 	checkServerIdentity: CheckServerIdentityFunction;
 }
@@ -1348,6 +1348,7 @@
 ];
 
 export default class Request extends Duplex implements RequestEvents<Request> {
+// @ts-ignore
 	['constructor']: typeof Request;
 
 	declare [kUnproxyEvents]: () => void;
@@ -1444,6 +1445,7 @@
 					options.body.destroy();
 				}
 
+				// @ts-ignore
 				this.destroy(error);
 				return;
 			}
@@ -1473,11 +1475,13 @@
 				}
 
 				// Queued writes etc.
+// @ts-ignore
 				for (const job of this[kJobs]) {
 					job();
 				}
 
 				// Prevent memory leak
+// @ts-ignore
 				this[kJobs].length = 0;
 
 				this.requestInitialized = true;
@@ -1489,6 +1493,7 @@
 
 				// This is a workaround for https://github.com/nodejs/node/issues/33335
 				if (!this.destroyed) {
+					// @ts-ignore
 					this.destroy(error);
 				}
 			}
@@ -2045,6 +2050,7 @@
 			try {
 				await Promise.all(promises);
 			} catch (error) {
+				// @ts-ignore
 				this._beforeError(error);
 				return;
 			}
@@ -2145,6 +2151,7 @@
 
 				await this._makeRequest();
 			} catch (error) {
+				// @ts-ignore
 				this._beforeError(error);
 				return;
 			}
@@ -2201,6 +2208,7 @@
 			await this._onResponseBase(response);
 		} catch (error) {
 			/* istanbul ignore next: better safe than sorry */
+			// @ts-ignore
 			this._beforeError(error);
 		}
 	}
@@ -2367,6 +2375,7 @@
 		// Fallback function
 		let fallbackFn: HttpRequestFunction;
 		if (options.http2) {
+			// @ts-ignore
 			fallbackFn = http2wrapper.auto;
 		} else {
 			fallbackFn = isHttps ? https.request : http.request;
@@ -2499,6 +2508,7 @@
 				throw new CacheError(error, this);
 			}
 
+			// @ts-ignore
 			throw new RequestError(error.message, error, this);
 		}
 	}
@@ -2510,6 +2520,7 @@
 				error = await hook(error);
 			}
 		} catch (error_) {
+			// @ts-ignore
 			error = new RequestError(error_.message, error_, this);
 		}
 
@@ -2575,6 +2586,7 @@
 						})
 					});
 				} catch (error_) {
+					// @ts-ignore
 					void this._error(new RequestError(error_.message, error_, this));
 					return;
 				}
@@ -2587,6 +2599,7 @@
 								await hook(this.options, typedError, retryCount);
 							}
 						} catch (error_) {
+							// @ts-ignore
 							void this._error(new RequestError(error_.message, error, this));
 							return;
 						}
--- a/source/create.ts
+++ b/source/create.ts
@@ -154,6 +154,7 @@
 				callInitHooks(defaults.options.hooks.init, options);
 				callInitHooks(options.hooks?.init, options);
 			} catch (error) {
+				// @ts-ignore
 				initHookError = error;
 			}
 
@@ -170,6 +171,7 @@
 			if (options.isStream) {
 				throw error;
 			} else {
+				// @ts-ignore
 				return createRejection(error, defaults.options.hooks.beforeError, options.hooks?.beforeError);
 			}
 		}
--- a/source/utils/deprecation-warning.ts
+++ b/source/utils/deprecation-warning.ts
@@ -7,7 +7,7 @@
 
 	alreadyWarned.add(message);
 
-	// @ts-expect-error Missing types.
+// @ts-ignore
 	process.emitWarning(`Got: ${message}`, {
 		type: 'DeprecationWarning'
 	});
